packages/teams-js/utils/validOrigins.json

```diff
@@ -0,0 +1,39 @@
+{
+  "validOrigins": [
```

Comment:
>validOrigins [](http://example.com/codeflow?start=3&length=12)

I am concerned that there are now two copies of this list, one in the code and one here. Would it be possible to keep the source of truth in this file and then import it into the JS so that the JS doesn't have to keep a copy of it? Then the fallback in the JS will be identical to what's on the CDN (at least at the time the JS package was built)?

Without that, keeping two lists will introduce the possibility of them getting out of sync.

I understand that you are operating under specific constrained timelines here, so if this suggestion will take a lot of time to implement, consider creating a WI for it to follow-up after you finish the feature. That said, I _think_ you can just easily import the json file like outlined [here](https://stackoverflow.com/a/50540064/7117308)

Reply:
This is done for a specific reason, I am using this PR to push up this list as an artifact for testing pulling it from a CDN, but I can't do that until this goes through. I am planning to handle that change in a separate PR once we get the results from this test.